src/components/SnackChat/SnackChat.js

```diff
@@ -242,114 +252,159 @@ class SnackChat extends Component {
-    );
-    this.ctx.restore();
-
-    // Clip face
-    const ears = this.averageBodyPosition.ears;
-    this.ctx.save();
-    this.ctx.translate(
-      ears.rightX + ears.width / 2,
-      ears.rightY + ears.height * ears.angle
-    );
-    this.ctx.rotate(ears.angle);
-    clipEllipse(this.ctx, 0, 0, ears.span * 1.5, ears.span * 1.5);
-    this.ctx.resetTransform();
-
-    // Re-draw face
-    this.drawBackground(video);
+    this.setState({
+      averageBodyPosition
+    });
 
     requestAnimationFrame(this.update);
   };
 
   onConnect = () => {
     this.ctx = this.canvas.current.getContext('2d');
-    this.ctx.lineWidth = 5;
-    this.ctx.strokeStyle = 'red';
-    this.filter = new Image();
-    this.filter.src = this.props.storeList[this.props.actualItem].image;
+    this.filter = this.props.storeList[this.props.actualItem].image;
     this.countdown();
     this.playGettingInPositionAnimation();
-    requestAnimationFrame(this.update);
   };
 
   onFail = () => {
     this.props.setSendWithPhoto(false);
     this.props.history.replace('/slackname');
   };
 
+  // render functions ================================================
+
+  renderHandsHeader = () => (
+    <div>
+      <div className="snackchat--header-text snackchat--header-text-left">
+        {this.state.counter >= PHOTO_ANIMATION_TIME && 'Taking photo in'}
+      </div>
+      <div className="snackchat--hands">
+        <img className="snackchat--hands-slot" src={HandsSlot} alt="" />
+        <img className="snackchat--hands-right" src={HandsRight} alt="" />
+        <img className="snackchat--hands-center" src={HandsCenter} alt="" />
+        <img className="snackchat--hands-camera" src={HandsCamera} alt="" />
+        <img className="snackchat--hands-left" src={HandsLeft} alt="" />
+      </div>
+      <div className="snackchat--header-counter">
+        {this.state.counter >= PHOTO_ANIMATION_TIME &&
+          this.state.counter - PHOTO_ANIMATION_TIME}
+      </div>
+    </div>
+  );
+
+  renderFallingItems = () => (
+    <div>
+      <div className="snackchat-overlay" />
+      {this.state.itemPositions.map((item, index) => (
+        <div key={index}>
+          <img
+            src={this.filter}
+            alt=""
+            className="snackchat-falling-snack"
+            style={{
+              height: FEED_SIZE * FALLING_SNACK_SIZE,
+              width: FEED_SIZE * FALLING_SNACK_SIZE,
+              top: item.y,
+              right: item.x - FEED_SIZE * 0.1,
+              transform: `rotate(${item.rotation}rad)`
+            }}
+          />
+        </div>
+      ))}
+    </div>
+  );
+
+  generateEllipseCoords = shoulders => {
+    const {ears} = this.state.averageBodyPosition;
+    const numberOfPoints = 40;
+    let ellipseCoords = '';
+    for (let i = 0; i <= numberOfPoints; ++i) {
+      ellipseCoords += `,${shoulders.span * 2 +
+        ears.span * 0.6 * Math.sin((i / numberOfPoints) * 2 * Math.PI)} ${0.9 *
+        shoulders.span +
```

Comment:
There's a few magic numbers in the calculations, do they have special meaning or do they just look right? Could we give them a name?

Again extracting the two variables would help make this a little easier to follow

```
for (let i = 0; i <= numberOfPoints; i++) {
  const angle = (i / numberOfPoints) * 2 * Math.PI
  const x = shoulders.span * 2 + ears.span * 0.6 * Math.sin(angle)
  const y = shoulders.span * 0.9 + ears.span * 0.75 + Math.cos(angle)

  ellipseCoords += `,${x} ${y}`
}
```

Since we can access the state directly, (and we are to get ears), we can get shoulders that way too instead of passing it as an argument

```
const { ears, shoulders } = this.state.averageBodyPosition
```

Or, if we'd prefer to not mutate and drop the trailing comma...

```
getEllipseCoords() {
  const { ears, shoulders } = this.state.averageBodyPosition
  const numberOfPoints = 40

  return Array.from({ length: numberOfPoints + 1 }).map((_, index) => {
    const angle = (index / numberOfPoints) * 2 * Math.PI
    const x = shoulders.span * 2 + ears.span * 0.6 * Math.sin(angle)
    const y = shoulders.span * 0.9 + ears.span * 0.75 + Math.cos(angle)

    return `${x} ${y}`
  }).join(',')
```

Reply:
wow that's so much nicer haha